move/move-ibc/sources/ibc.move

```diff
@@ -1156,22 +1173,34 @@ module ibc::ibc {
 
     // ========= UTILS and VIEW functions ========= //
 
+    #[view]
+    public fun client_id_to_type(client_id: u32): String acquires IBCStore {
+        let store = borrow_global<IBCStore>(get_vault_addr());
+        *smart_table::borrow(&store.client_id_to_type, client_id)
+    }
+
     #[view]
     public fun get_module(channel_id: u32): address acquires IBCStore {
         let store = borrow_global<IBCStore>(get_vault_addr());
         *smart_table::borrow(&store.channel_to_module, channel_id)
     }
 
     #[view]
-    public fun client_state(client_type: String, client_id: u32): vector<u8> {
-        light_client::get_client_state(client_type, client_id)
+    public fun client_state(client_id: u32): vector<u8> acquires IBCStore {
+        let store = borrow_global<IBCStore>(get_vault_addr());
+        light_client::get_client_state(
+            *smart_table::borrow(&store.client_id_to_type, client_id), client_id
```

Comment:
can `client_id_to_type` not be used here?

Reply:
also, client id should be a global counter - the type shouldn't be needed to fetch the state?

Reply:
we have to fetch the client type because we call the correct light client implementation based on it. Dynamic dispatch is not implemented for the light clients.

Reply:
ah of course